ecommerce/pricing/lib/pricing/commands.rb

```diff
@@ -45,6 +52,16 @@ class RegisterCoupon < Infra::Command
     alias aggregate_id coupon_id
   end
 
+  class CreateHappyHour < Infra::Command
+    attribute :id, Infra::Types::UUID.meta(omittable: true)
+    attribute :product_ids, Infra::Types::Array.of(Infra::Types::UUID)
+    attribute :name, Infra::Types::String
+    attribute :code, Infra::Types::String
+    attribute :discount, Infra::Types::HappyHourDiscount
+    attribute :start_hour, Infra::Types::Hour
+    attribute :end_hour, Infra::Types::Hour
+  end
```

Comment:
I didn't think about it much yet - but this command looks "big" and probably reflects the UI, which is often a smell. 

If in the future we split the UI into several screens and provide partial data, the command would no longer work so nicely.

The idea of commands is to keep them decoupled from the current UI. Usually this means, to have them mush smallem - 2-3 attributes max.

It may not always be best idea, just a rule of thumb.

Reply:
Not sure if this helps, but I moved most of the attributes to a custom type - `HappyHourDetails`. Is that solution you were thinking about?